libs/webb-ui-components/src/components/SideBar/components/Item.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { useEffect, useMemo, useState } from 'react';
+import { DropdownControl } from './DropdownControl';
+import { SubItem, SubItemProps } from './SubItem';
+import { ExternalLinkLine } from '@webb-tools/icons';
+import { pushToInternalLink, pushToExternalLink } from '../utils';
+import { IconBase } from '@webb-tools/icons/types';
+import { Typography } from '../../../typography/Typography';
+import { twMerge } from 'tailwind-merge';
+
+export type ItemProps = {
+  name: string;
+  isInternal: boolean;
+  href: string;
+  Icon: (props: IconBase) => JSX.Element;
+  subItems: SubItemProps[];
+};
+
+export type ExtraItemProps = {
+  isSidebarOpen?: boolean;
+  isActive?: boolean;
+  setIsActive?: () => void;
+};
+
+export const Item: React.FC<ItemProps & ExtraItemProps> = ({
+  name,
+  isInternal,
+  href,
+  Icon,
+  subItems,
+  isSidebarOpen,
+  isActive,
+  setIsActive,
+}) => {
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+  const [activeSubItem, setActiveSubItem] = useState<number | null>(null);
+
+  useEffect(() => {
+    if (isActive) {
+      setIsDropdownOpen(true);
+    } else {
+      setIsDropdownOpen(false);
+    }
+  }, [isActive]);
+
+  const pushToLinkAndToggleDropdown = () => {
+    if (!isSidebarOpen && setIsActive && isInternal) {
+      setIsActive();
+    }
+
+    if (!isSidebarOpen && isInternal && href) {
+      pushToInternalLink(href);
+    } else if (!isInternal && href) {
+      pushToExternalLink(href);
+    }
```

Comment:
I think this is not really good for accessibility, we should use the anchor tag `<a></a>` for links instead of `window.open` here.